statsmodels/gam/gam_cross_validation/gam_cross_validation.py

```diff
@@ -0,0 +1,170 @@
+import itertools
+import numpy as np
+from statsmodels.gam.smooth_basis import (GenericSmoothers,
+                                          UnivariateGenericSmoother)
+
+try:
+    import matplotlib.pyplot as plt
+    have_matplotlib = True
+except ImportError:
+    have_matplotlib = False
+
+
+class BaseCV(with_metaclass(ABCMeta)):
+    """
+    BaseCV class. It computes the cross validation error of a given model.
+    All the cross validation classes can be derived by this one
+    (e.g. GamCV, LassoCV,...)
+    """
+
+    def __init__(self, cv, x, y):
+        self.cv = cv
+        self.x = x
+        self.y = y
+        self.train_test_cv_indices = self.cv.split(self.x, self.y, label=None)
+
+    def fit(self, **kwargs):
+        # kwargs are the input values for the fit method of the
+        # cross-validated object
+
+        cv_err = []
+
+        for train_index, test_index in self.train_test_cv_indices:
+            cv_err.append(self._error(train_index, test_index, **kwargs))
+
+        return np.array(cv_err)
+
+    @abstractmethod
+    def _error(self, train_index, test_index, **kwargs):
+        # train the model on the train set and returns the error on the test set
+        pass
+
+
+def _split_train_test_smoothers(x, smoothers, train_index, test_index):
+    train_smoothers = []
+    test_smoothers = []
+    for i, smoother in enumerate(smoothers.smoothers):
+        train_basis = smoother.basis[train_index]
+        train_der_basis = smoother.der_basis[train_index]
+        train_der2_basis = smoother.der2_basis[train_index]
+        train_cov_der2 = smoother.cov_der2
+        # TODO: Double check this part. cov_der2 is calculated with all the data
+        train_x = smoother.x[train_index]
+
+        train_smoothers.append(UnivariateGenericSmoother(train_x, train_basis,
+            train_der_basis, train_der2_basis, train_cov_der2,
+            smoother.variable_name + ' train'))
+
+        test_basis = smoother.basis[test_index]
+        test_der_basis = smoother.der_basis[test_index]
+        test_der2_basis = smoother.der2_basis[test_index]
+        test_cov_der2 = smoother.cov_der2
+        # TODO: Double check this part. cov_der2 is calculated with all the data
+        test_x = smoother.x[test_index]
+
+        test_smoothers.append(UnivariateGenericSmoother(test_x, test_basis,
+            test_der_basis, train_der2_basis, test_cov_der2,
+            smoother.variable_name + ' test'))
+
+    train_multivariate_smoothers = GenericSmoothers(x[train_index],
+                                                    train_smoothers)
+    test_multivariate_smoothers = GenericSmoothers(x[test_index],
+                                                   test_smoothers)
+
+    return train_multivariate_smoothers, test_multivariate_smoothers
+
+
+class MultivariateGAMCV(BaseCV):
+    def __init__(self, smoothers, alphas, gam, cost, y, cv):
+        # the gam class has already an instance
```

Comment:
can you flesh this comment out?  im confused in several directions

Reply:
"TODO: figure out what the end of the next sentence is supposed to be"?  if it isn't clear to you, its definitely not clear to the reader

Reply:
I delete the comment as misleading here